examples/pixart_sigma/pixart/pipelines/infer_pipeline.py

```diff
@@ -0,0 +1,173 @@
+
+from mindone.diffusers import AutoencoderKL
+from mindone.transformers import T5EncoderModel
+
+
+class PixArtInferPipeline:
+    """
+    Args:
+        network (nn.Cell): `PixArt` network to denoise the encoded image latents.
+        vae (nn.Cell): Variational Auto-Encoder (VAE) Model to encode and decode images to and from latent representations.
+        scale_factor (float): scale_factor for vae.
+        guidance_scale (float): A higher guidance scale value for noise rescale.
+        num_inference_steps: (int): The number of denoising steps.
+    """
+
+    def __init__(
+        self,
+        network: PixArt,
+        vae: AutoencoderKL,
+        text_encoder: T5EncoderModel,
+        text_tokenizer: AutoTokenizer,
+        scale_factor: float = 1.0,
+        guidance_scale: float = 0.0,
+        num_inference_steps: int = 100,
+        sampling_method: Literal["iddpm", "ddim", "dpm"] = "iddpm",
+        force_freeze: bool = False,
+    ):
+        super().__init__()
+        self.network = network
+        self.vae = vae
+        self.text_encoder = text_encoder
+        self.text_tokenizer = text_tokenizer
+        self.scale_factor = scale_factor
+        self.guidance_scale = Tensor(guidance_scale, dtype=ms.float32)
+        self.sampling_method = sampling_method
+        self.num_inference_steps = num_inference_steps
+
+        if sampling_method == "iddpm":
+            self.diffusion = create_diffusion(str(num_inference_steps))
+            self.sampling_func = self.diffusion.p_sample_loop
+        elif sampling_method == "ddim":
+            self.diffusion = create_diffusion(str(num_inference_steps))
+            self.sampling_func = self.diffusion.ddim_sample_loop
+        else:
+            self.noise_schedule = create_noise_schedule_dpms()
+            self.diffusion = partial(
+                DPMS, model=self.network.construct_with_dpmsolver, noise_schedule=self.noise_schedule
+            )
+            self.sampling_func = None
+
+        if force_freeze:
+            # freeze all components
+            self.network.set_train(False)
+            for param in self.network.trainable_params():
+                param.requires_grad = False
+
+            self.vae.set_train(False)
+            for param in self.vae.trainable_params():
+                param.requires_grad = False
+
+            self.text_encoder.set_train(False)
+            for param in self.text_encoder.trainable_params():
+                param.requires_grad = False
+
+    @ms.jit
+    def vae_decode(self, x):
+        """
+        Args:
+            x: (b c h w), denoised latent
+        Return:
+            y: (b H W 3), batch of images, normalized to [0, 1]
+        """
+        y = self.vae.decode(x / self.scale_factor)[0]
+        y = ops.clip_by_value((y + 1.0) / 2.0, clip_value_min=0.0, clip_value_max=1.0)
+
+        # (b 3 H W) -> (b H W 3)
+        y = ops.transpose(y, (0, 2, 3, 1))
+
+        return y
+
+    @ms.jit
```

Comment:
why add jit?

Reply:
it is faster in batch inference in pynative mode, no effect in graph mode